HGustavs/LenaSYS
# Implemented db_setup file

### Implemented db_setup
Pull request contains db_setup file with the following functions:
- Create database
- Create user
- Delete database
- Delete user
- Set permissions for user on database
- run sql file
- verbose mode for running sql file

### How to test?

Append the code below to the file `newinstaller/installer.php`. 
Change the pdo declaration to your mysql root credentials. 
This will modify your database, so ensure that you dont accidentally delete anything.
Also make sure that you try to change some of the settings to ensure that it all works properly. You could for example test the verbose mode for the function `execute_sql_file()`.

```php
try {
    $hostname = 'db';
    $username = 'root';
    $password = 'password';
    
    $dsn = "mysql:host=$hostname;charset=utf8mb4";
    $pdo = new PDO($dsn, $username, $password);
    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
    echo "Connected successfully to MySQL server" . "<br>";
} catch (PDOException $e) {
    echo "Connection failed: " . $e->getMessage();
}

$installer = new DBSetup(pdo: $pdo, db_name: "LenaSYStest", db_user: "LenaTest", db_user_password: "test", hostname:"%");
echo $installer->create_db()['message'] . "<br>";
echo $installer->create_user(force: true)['message'] . "<br>";
echo $installer->set_permissions()['message'] . "<br>";

ob_start();
function callback($message) {
    echo $message . "<br>";
    ob_flush();
    flush();
}
echo $installer->execute_sql_file("../Shared/SQL/init_db.sql", "callback", verbose: false)["message"] . "<br>";

echo $installer->drop_user()['message'] . "<br>";
echo $installer->drop_db()['message'] . "<br>";
```
The output should look like this:
![image](https://github.com/HGustavs/LenaSYS/assets/129264968/ffc4d4b6-c3cf-436e-89ad-c0a55b6625aa)


closes #15154
